### PR TITLE
[JR-1170] Validation of location for EU countries should only check not null/constraint

### PR DESCRIPTION
**Pull request by wwebec**

### CODE REVIEW

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java [hunk 1]
```diff
@@ -48,17 +48,17 @@ public Location enrichCodes(Location location) {
     }
 
     @Override
-    public boolean verifyLocation(Location location) {
-        if (isManagedLocations(location)) {
-            return locationApiClient.findLocationByPostalCodeAndCity(location.getPostalCode(), location.getCity()).isPresent();
+    public boolean isLocationValid(Location location) {
```
Comment (cjpsycos) [inline infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: The only task was to rename the method to isLocationValid and change the default return result to true.
Now the method has a totally other behavior.
Please revert this change and just do the two task

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java [hunk 2]
```diff
@@ -48,17 +48,17 @@ public Location enrichCodes(Location location) {
...
-
-        return false;
+        return isManagedLocations(location) ?
+                hasText(location.getPostalCode()) :
+                true;
     }
 
     private boolean isManagedLocations(Location location) {
         return location != null
-                && hasText(location.getPostalCode())
```
Comment (cjpsycos) [inline infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: Revert this line

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java (hunk 1 above)
Comment (wwebec) [reply infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: A location is valid if it has one of manged ISO codes CH or LI and has postal code and can be retrieved (is present in db).
Alternatively if location is not empty it is not validated at all, it has only be not empty.

For that reason current implementation was not sufficient. Could you please verify? Tests illustrates above behavior.

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java (hunk 2 above)
Comment (wwebec) [reply infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: please see answer to the previous comment.

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java [hunk 3]
```diff
@@ -48,17 +48,17 @@ public Location enrichCodes(Location location) {
...
-        if (isManagedLocations(location)) {
-            return locationApiClient.findLocationByPostalCodeAndCity(location.getPostalCode(), location.getCity()).isPresent();
+    public boolean isLocationValid(Location location) {
+        if (location == null) {
+            return false;
         }
-
-        return false;
+        return isManagedLocations(location) ?
+                hasText(location.getPostalCode()) && locationApiClient.findLocationByPostalCodeAndCity(location.getPostalCode(), location.getCity()).isPresent() :
```
Comment (cjpsycos) [inline infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: extract postalcode check and findlocation in private method

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java [hunk 4]
```diff
@@ -48,17 +48,17 @@ public Location enrichCodes(Location location) {
     }
```
Comment (cjpsycos) [inline infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: check after isManageLocations if a postalCode is present. if not the return location.
if(!hasText(location.postalCode) then return location.

Comment (wwebec) [reply infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: added a postal check for ```enrichCodes``` and respective tests

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java [hunk 5]
```diff
@@ -41,24 +42,28 @@ public Location enrichCodes(Location location) {
...
+    }
 
-        return false;
+    private Optional<LocationResource> findLocationIfHasPostalCode(Location location) {
+        return hasText(location.getPostalCode()) ?
+                locationApiClient.findLocationByPostalCodeAndCity(location.getPostalCode(), location.getCity())
+                : Optional.empty();
     }
 
     private boolean isManagedLocations(Location location) {
```
Comment (pavax) [inline infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: rename to isMangedLocation

File: infrastructure-service-reference/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationServiceTest.java [hunk 6]
```diff
@@ -109,4 +111,87 @@ public void shouldNotEnrichCodesWhenNoMatchForPostalCodeAndCity() {
...
+                .setCity("Liechtenstein")
+                .build();
+        // WHEN
+        Location enriched = testedObject.enrichCodes(location);
+
+        assertThat(enriched).isSameAs(location);
+    }
+
+    @Test
+    public void shouldNotEnrichCodesIfLocationIsSwityerlandAndHasNotPostalCodee() {
```
Comment (pavax) [inline infrastructure-service-reference/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationServiceTest.java]: typo: Switzerland

File: infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java (hunk 5 above)
Comment (cjpsycos) [reply infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: isManagedLocation

Comment (wwebec) [reply infrastructure-service-reference/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationService.java]: renamed

File: infrastructure-service-reference/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationServiceTest.java (hunk 6 above)
Comment (wwebec) [reply infrastructure-service-reference/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/service/reference/location/DefaultLocationServiceTest.java]: renamed
